### PR TITLE
Added OpenRouter support

### PR DESCRIPTION
**Pull request by zakhar-kogan**
# Features
- Added [OpenRouter](https://openrouter.ai/) support as it gives access to GPT-4, PaLM-2 and Claude 2 without additional surcharges;
- Added support for .env files and enclosed an example.

### CODE REVIEW

**Comment by ghost** [comment]
PR under review ✅ 

Hope to revert by tonight.

File: completion_test.py [hunk 1]
```diff
@@ -8,23 +8,27 @@
...
-print("\nDavinci call")
-print(response)
-
-# cohere call
-response = completion("command-nightly", messages)
-print("\nCohere call")
-print(response)
-
+# # openai call
+# response = completion(model="gpt-3.5-turbo", messages=messages)
```
Comment (ishaan-jaff) [inline completion_test.py]: can we leave all the other tests uncommented

**Review by ishaan-jaff — COMMENTED** [review]
thanks for the contribution, looks awesome. It's good to merge, but small nit: In completion_test can you leave all the other tests uncommented

File: completion_test.py (hunk 1 above)
Comment (zakhar-kogan) [reply completion_test.py]: Will do in ~15 minutes. Thanks for reviewing 🧡

Comment (zakhar-kogan) [reply completion_test.py]: Uncommented + deleted the unneeded testos.py (my bad) 🥲

One question - I could try implementing everything using pattern matching. Should I try that, or we'll lose backward compatibility this way?

Comment (ishaan-jaff) [reply completion_test.py]: let's address pattern matching in a separate PR

**Pull request merged**